commands/pull.go

```diff
@@ -81,22 +82,32 @@ func (c *singleCheckout) Run(p *lfs.WrappedPointer) {
 		return
 	}
 
+	if !c.RunToPath(p, cwdfilepath) {
+		return
+	}
+
+	// errors are only returned when the gitIndexer is starting a new cmd
+	if err := c.gitIndexer.Add(cwdfilepath); err != nil {
+		Panic(err, "Could not update the index")
+	}
+}
+
+// RunToPath checks out the pointer specified by p to the given path and returns
+// a boolean indicating whether it was successful.  It does not perform any sort
+// of sanity checking or add the path to the index.
+func (c *singleCheckout) RunToPath(p *lfs.WrappedPointer, path string) bool {
```

Comment:
I wonder if as a preparatory patch (or series) you might want to change the return type from `bool` to `error`. It may not be wise, since I believe that the implementation below already handles the error (i.e., sends the appropriate `pkt-line`, or similar), but I'm not sure.

It might be useful for callers to have the information, anyway.

Reply:
This commit is new, so I can make it `error` instead of `bool`.  The only reason I had to make it not return nothing is because we need to return early in `Run` in certain cases.